supabase/migrations/30_manual_billing.sql

```diff
@@ -0,0 +1,352 @@
+begin;
+
+create table internal.manual_bills (
+  tenant       catalog_tenant not null references tenants(tenant),
+  usd_cents    integer        not null,
+  description  text           not null,
+  date_start   date           not null,
+  date_end     date           not null,
+  constraint dates_make_sense check (date_start < date_end),
+  primary key (tenant, date_start, date_end)
+);
+
+comment on table internal.manual_bills is
+  'Manually entered bills that span an arbitrary date range';
+
+-- Move billing report gen to internal
+
+-- Drop the public functions
+drop function billing_report_202308(catalog_prefix, timestamptz);
+drop function tier_line_items(integer, integer[], text, text);
+
+-- Move `billing_historicals` to internal and revoke access
+alter table billing_historicals disable row level security;
+drop policy "Users must be authorized to their catalog tenant" on billing_historicals;
+
+alter table billing_historicals set schema internal;
+
```

Comment:
do you need a `revoke select on billing_historicals from authenticated`?

Reply:
Do I? I thought that `authenticated` doesn't have access to the `internal` schema at all, and that moving to `internal` would implicitly remove that grant (or make it impossible to address because `internal` "doesn't exist" if you aren't granted to it). But it doesn't hurt to be explicit, I'll add it.

Reply:
that's why i asked: I don't know what happens. I know it's currently granted and i don't know why moving it to a different schema would necessarily remove the grant.

Reply:
Turns out, the answer is "it would have worked correctly"

```sql
create table test(col integer primary key);
grant select on test to authenticated;
alter table test set schema internal;
set role authenticated;
select * from internal.test;
```
=>
```
ERROR:  permission denied for schema internal
LINE 1: select * from internal.test;
```